chain-server/src/server.rs

```diff
@@ -173,13 +176,13 @@ where
                     &result,
                     &next_callable,
                     &self.store,
-                    proof,
+                    compressed_proof,
```

Comment:
So, while this works, the particular interaction we want is that we have K > 1 new inputs for one (compressed output). Currently, I believe this is limited to K = 1.

There's two ways to go about this: 
- the client provides a list of K inputs, receives a compressed output,
- the client provides inputs as they wish, and has one command that allows them to request, with no further input, a compressed proof for everything up to that point.

This doesn't have to be resolved here, but we probably want to open an issue.